### ISSUE TITLE
System Tray Icons

### ISSUE THREAD

**Issue by DavidRaveloU** [amnweb/yasb#30]
**What is the request?.**
I would like to request the addition of an option to view and interact with the hidden icons in the System Tray. 

**Why is it useful?**
This feature would make it more convenient to access and manage the icons that are currently hidden in the System Tray.

**How could it be implemented?**
One possible implementation could be to add a toggle or expandable section in the System Tray that reveals all hidden icons and makes them clickable

**Comment by amnweb** [comment]
That would be nice to have but I don't have any idea how to access the system tray on Windows 11. We talked about this on Discord and if I can remember someone has found a solution for Windows 10 but Windows 11 is problematic. I will pin this issue maybe someone can help with this.

**Comment by DavidRaveloU** [comment]
I remember that feature being available in a Steam program called MyDockFinder, which lets you access and interact with System Tray icons. Here’s the direct link:

https://store.steampowered.com/app/1787090/MyDockFinder/

Unfortunately, it’s not open source, so I don’t know how it could be done.

**Comment by im-izz** [comment]
> That would be nice to have but I don't have any idea how to access the system tray on Windows 11. We talked about this on Discord and if I can remember someone has found a solution for Windows 10 but Windows 11 is problematic. I will pin this issue maybe someone can help with this.

How its done for windows 10? im running windows 10 and would love to see my tray icons there

**Comment by Zevan770** [comment]
An ugly "solution" is to use [translucentTB](https://github.com/TranslucentTB/TranslucentTB) to make system taskbar transparent and then overlap the system status bar with the yasb bar.

In fact it shouldn't even be called a solution, just happens to fit my needs of displaying komorebi workspaces using yasb without losing the system taskbar's information, including the tray icons.

What it looks like:
![image](https://github.com/user-attachments/assets/033a793c-1e36-46cc-92bf-8df4a997ff8e)

shortcoming:
yasb cannot accept mouse events.

好吧确实非常ugly，但是

**Comment by Osyx** [comment]
I saw that Seelen UI has got them working, using this [call](https://github.com/eythaann/Seelen-UI/blob/master/src/background/modules/tray/application.rs#L55), which does it differently depending on win 10/11, like you said. 

It is in rust, but tells us that 1. it is possible, 2. for Win 11 you can use `Windows.UI.Composition.DesktopWindowContentBridge` to access the tasks via the handle `TopLevelWindowForOverflowXamlIsland`. 

Might help somehow!

**Comment by Video-Nomad** [comment]
Looks like this is exactly how this should be implemented with the "ghost" window intercepting the messages and forwarding them to the real systray after we got all the data we need from it for the systray widget.
https://github.com/glzr-io/zebar/issues/57#issuecomment-2359183981
https://github.com/glzr-io/zebar/pull/192
https://github.com/glzr-io/zebar/blob/main/crates/systray-util/README.md

**Comment by amnweb** [comment]
> Looks like this is exactly how this should be implemented with the "ghost" window intercepting the messages and forwarding them to the real systray after we got all the data we need from it for the systray widget. [glzr-io/zebar#57 (comment)](https://github.com/glzr-io/zebar/issues/57#issuecomment-2359183981) [glzr-io/zebar#192](https://github.com/glzr-io/zebar/pull/192) https://github.com/glzr-io/zebar/blob/main/crates/systray-util/README.md

Do you have any examples in Python? Some things I can't understand there

**Comment by Video-Nomad** [comment]
> > Looks like this is exactly how this should be implemented with the "ghost" window intercepting the messages and forwarding them to the real systray after we got all the data we need from it for the systray widget. [glzr-io/zebar#57 (comment)](https://github.com/glzr-io/zebar/issues/57#issuecomment-2359183981) [glzr-io/zebar#192](https://github.com/glzr-io/zebar/pull/192) https://github.com/glzr-io/zebar/blob/main/crates/systray-util/README.md
> 
> Do you have any examples in Python? Some things I can't understand there

No. Haven't had the chance to test any of it yet, just the general idea.
